lib/kernel/src/trace.erl

```diff
@@ -0,0 +1,1229 @@
+  [`spawned`](#process_trace_messages_spawned),
+  [`exit`](#process_trace_messages_exit),
+  [`register`](#process_trace_messages_register),
+  [`unregister`](#process_trace_messages_unregister),
+  [`link`](#process_trace_messages_link),
+  [`unlink`](#process_trace_messages_unlink),
+  [`getting_linked`](#process_trace_messages_getting_linked), and
+  [`getting_unlinked`](#process_trace_messages_getting_unlinked).
+
+- **`running`** - Traces scheduling of processes.
+
+  Message tags: [`in`](#process_trace_messages_in_proc) and
+  [`out`](#process_trace_messages_out_proc).
+
+- **`exiting`** - Traces scheduling of exiting processes.
+
+  Message tags:
+  [`in_exiting`](#process_trace_messages_in_exiting_proc),
+  [`out_exiting`](#process_trace_messages_out_exiting_proc), and
+  [`out_exited`](#process_trace_messages_out_exited_proc).
+
+- **`running_procs`** - Traces scheduling of processes just like `running`.
+  However, this option also includes schedule events when the process executes
+  within the context of a port without being scheduled out itself.
+
+  Message tags: [`in`](#process_trace_messages_in_proc) and
+  [`out`](#process_trace_messages_out_proc).
+
+- **`garbage_collection`** - Traces garbage collections of processes.
+
+  Message tags:
+  [`gc_minor_start`](#process_trace_messages_gc_minor_start),
+  [`gc_max_heap_size`](#process_trace_messages_gc_max_heap_size), and
+  [`gc_minor_end`](#process_trace_messages_gc_minor_end).
+
+- **`timestamp`{: #timestamp }** - Includes a time stamp in all trace
+  messages. The time stamp (Ts) has the same form as returned by
+  `erlang:now()`.
+
+- **`cpu_timestamp`** - A global trace flag for the Erlang node that makes all
+  trace time stamps using flag `timestamp` to be in CPU time, not wall clock
+  time. That is, `cpu_timestamp` is not be used if `monotonic_timestamp` or
+  `strict_monotonic_timestamp` is enabled. Only allowed with `PidPortSpec==all`.
+  If the host machine OS does not support high-resolution CPU time measurements,
+  [`trace/3`](`trace/3`) exits with `badarg`. Notice that most OS do not
+  synchronize this value across cores, so be prepared that time can seem to go
+  backwards when using this option.
+
+- **`monotonic_timestamp`** - Includes an
+  [Erlang monotonic time](`e:erts:time_correction.md#erlang-monotonic-time`) time stamp
+  in all trace messages. The time stamp (Ts) has the same format and value as
+  produced by [`erlang:monotonic_time(nanosecond)`](`monotonic_time/1`). This
+  flag overrides flag `cpu_timestamp`.
+
+- **`strict_monotonic_timestamp`** - Includes an time stamp consisting of
+  [Erlang monotonic time](`e:erts:time_correction.md#erlang-monotonic-time`) and a
+  monotonically increasing integer in all trace messages. The time stamp (Ts)
+  has the same format and value as produced by `{`
+  [`erlang:monotonic_time(nanosecond)`](`monotonic_time/1`)`,`
+  [`erlang:unique_integer([monotonic])`](`unique_integer/1`)`}`. This flag
```

Comment:
```suggestion
  [`erlang:monotonic_time(nanosecond)`](`erlang:monotonic_time/1`)`,`
  [`erlang:unique_integer([monotonic])`](`erlang:unique_integer/1`)`}`. This flag
```

Reply:
Is this by design, that it's ok to write \`foo/1\` for a function that does not exists in the current module?

Reply:
I see now I got warnings like this:
```
    warning: documentation references function "monotonic_time/1" but it is undefined or private
```

Reply:
Yes it is by design. To guarantee that you get a warning the link to a local function has to be via an explicit link and not via auto linking, that is ``[foo/1](`foo/1`)`` instead of just `foo/1`. There are way too many false positives in docs otherwise.